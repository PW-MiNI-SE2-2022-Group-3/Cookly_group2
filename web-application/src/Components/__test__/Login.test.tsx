import React from "react";
import ReactDOM from "react-dom";
import {
  render,
  screen,
  fireEvent,
  queryByTestId,
  cleanup,
} from "@testing-library/react";
import LoginScreen from "./../Login";
import { isTSAnyKeyword } from "@babel/types";

afterEach(cleanup);

it("renders correctly without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<LoginScreen setIsLogged={false} />, div);
});

it("renders logo correctly", () => {
  const { getByTestId } = render(<LoginScreen setIsLogged={false} />);
  expect(getByTestId("logo-img")).toHaveAttribute("src", "logo.png");
  expect(getByTestId("logo-img")).toHaveAttribute("alt", "logo");
  expect(getByTestId("logo-img")).toHaveAttribute("width", "300");
});

it("renders background correctly", () => {
  const { getByTestId } = render(<LoginScreen setIsLogged={false} />);
  expect(getByTestId("background-img")).toHaveAttribute("src", "bg1.jpg");
  expect(getByTestId("background-img")).toHaveAttribute("alt", "brand");
});

it("renders button correctly", () => {
  const { getByTestId } = render(<LoginScreen setIsLogged={false} />);
  expect(getByTestId("login-button")).toBeInTheDocument();
  expect(getByTestId("login-button")).toHaveTextContent("Log In");
});

it("renders input fields correctly", () => {
  const { getByTestId } = render(<LoginScreen setIsLogged={false} />);
  expect(getByTestId("username-textfield")).toBeInTheDocument();
  expect(getByTestId("password-textfield")).toBeInTheDocument();
});

it("matches snapshot", () => {
  const tree = render(<LoginScreen setIsLogged={false} />);
  expect(tree).toMatchSnapshot();
});
