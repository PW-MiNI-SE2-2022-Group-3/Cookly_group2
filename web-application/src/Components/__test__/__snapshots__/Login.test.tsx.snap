// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="MuiGrid-root MuiGrid-container"
          style="min-height: 100vh;"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
          >
            <img
              alt="brand"
              data-testid="background-img"
              src="bg1.jpg"
              style="width: 100%; height: 100%; object-fit: cover;"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
            style="padding: 10px;"
          >
            <div />
            <div
              style="display: flex; flex-direction: column; max-width: 400px; min-width: 300px;"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-center"
              >
                <img
                  alt="logo"
                  data-testid="logo-img"
                  src="logo.png"
                  width="300"
                />
              </div>
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
                data-testid="username-textfield"
              >
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
                >
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionStart css-ittuaa-MuiInputAdornment-root"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="AccountCircleIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                      />
                    </svg>
                  </div>
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
                    id="username"
                    placeholder="Username"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
                data-testid="password-textfield"
              >
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
                >
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionStart css-ittuaa-MuiInputAdornment-root"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="LockRoundedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zM9 8V6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9z"
                      />
                    </svg>
                  </div>
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
                    id="password"
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </div>
              </div>
              <div
                style="height: 20px;"
              />
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained"
                data-testid="login-button"
                id="login-button"
                placeholder="Log In"
                style="background-color: rgb(196, 86, 12); color: white; border-radius: 0;"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Log In
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container"
        style="min-height: 100vh;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <img
            alt="brand"
            data-testid="background-img"
            src="bg1.jpg"
            style="width: 100%; height: 100%; object-fit: cover;"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
          style="padding: 10px;"
        >
          <div />
          <div
            style="display: flex; flex-direction: column; max-width: 400px; min-width: 300px;"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-center"
            >
              <img
                alt="logo"
                data-testid="logo-img"
                src="logo.png"
                width="300"
              />
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
              data-testid="username-textfield"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
              >
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionStart css-ittuaa-MuiInputAdornment-root"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="AccountCircleIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                    />
                  </svg>
                </div>
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
                  id="username"
                  placeholder="Username"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
              data-testid="password-textfield"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
              >
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionStart css-ittuaa-MuiInputAdornment-root"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="LockRoundedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zM9 8V6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9z"
                    />
                  </svg>
                </div>
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
                  id="password"
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </div>
            </div>
            <div
              style="height: 20px;"
            />
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained"
              data-testid="login-button"
              id="login-button"
              placeholder="Log In"
              style="background-color: rgb(196, 86, 12); color: white; border-radius: 0;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Log In
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
